crates/core/src/nats/nats_conn.rs

```diff
@@ -1,23 +1,52 @@
-use super::{ConnStreams, NatsClient, NatsError};
+use super::{types::*, ClientOpts, ConnId, ConnStreams, NatsClient, NatsError};
 
 #[derive(Debug, Clone)]
 pub struct NatsConn {
-    pub client: NatsClient,
-    pub streams: ConnStreams,
+    client: NatsClient,
+    streams: ConnStreams,
 }
 
 impl NatsConn {
-    pub async fn new(
-        conn_id: &str,
-        nats_url: &str,
-        nats_nkey: &str,
-    ) -> Result<Self, NatsError> {
-        let client = NatsClient::connect(nats_url, conn_id, nats_nkey).await?;
+    #[cfg(feature = "test_helpers")]
+    pub async fn connect(opts: ClientOpts) -> Result<Self, NatsError> {
+        let client = NatsClient::connect(opts).await?;
         let streams = ConnStreams::new(&client).await?;
 
         Ok(Self {
             streams,
             client: client.clone(),
         })
     }
+
+    pub fn opts(&self) -> ClientOpts {
+        self.client().opts.clone()
+    }
+
+    pub fn conn_id(&self) -> ConnId {
+        self.client().opts.conn_id.clone()
+    }
+
+    pub fn url(&self) -> String {
+        self.client().opts.url
+    }
+
+    pub fn client(&self) -> NatsClient {
+        self.client.clone()
+    }
+
+    pub fn state(&self) -> ConnectionState {
+        self.nats_client().connection_state()
+    }
+
+    pub fn nats_client(&self) -> AsyncNatsClient {
+        self.client().nats_client.clone()
+    }
+
+    pub fn jetstream(&self) -> JetStreamContext {
+        self.client().jetstream.clone()
+    }
+
+    pub fn streams(&self) -> ConnStreams {
+        self.streams.clone()
+    }
```

Comment:
Most of these fields are static and can be normalized on `::new()` -- we can then simplify by making them public. Something like:

```rust
   pub struct NatsConn {
        pub client: NatsClient,
        pub conn_id: ConnId,
        pub url: string,
        pub nats_client: AsyncNatsClient,
        pub jetstream: JetStreamContext,
        pub streams: ConnStreams,
    }
```

Reply:
With our `semver` versioning, we shouldn't have to worry about encapsulation-related breaking-change problems for static fields like these ones